doc/rst/language/evolution.rst

```diff
@@ -47,32 +47,200 @@ Description forthcoming.
+  writeln(p:Child); // this downcast fails and will throw a ClassCastError
+
+casting C pointers to classes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Casts from ``c_void_ptr`` to class types were previously allowed. However,
+since ``c_void_ptr`` can store ``NULL``, this case needs adjustment
+following the nilability changes. Additionally, since ``c_void_ptr``
+refers to a C pointer, and C pointers are manually managed (i.e. you call
+``free`` on them at the appropriate time), it makes the most sense
+for casts from ``c_void_ptr`` to end up with an unmanaged type.
+
+Consider the following example:
+
+.. code-block:: chapel
+
+  class C {
+    var x:int;
+  }
+
+  var myC = new owned C();
+  var ptr:c_void_ptr = myC.borrow(); // store the instance in a C ptr
+
+Now we can cast from ``ptr`` to the class type:
+
+.. code-block:: chapel
+
+  var c = ptr:C; // cast from a C pointer to the borrowed type
+
+This example would work in 1.19. In 1.20, it needs to be updated to
+cast to ``unmanaged C?``:
+
+.. code-block:: chapel
+
+  var c = ptr:unmanaged C?;
+
+As with other values of type ``unmanaged C?``, from there it can:
+
+ * be borrowed, e.g. ``c.borrow()``
+ * have ``!`` applied to convert to a non-nilable value or halt, e.g. ``c!``
+ * be cast to a nilable type, throwing if it is ``nil``, e.g. ``c:borrowed C``
 
 .. _readme-evolution.undecorated-classes-generic-management:
 
 undecorated classes have generic management
 ********************************************
 
-Description forthcoming.
+Undecorated classes now have generic management. As an aid in migrating
+code to this change, the flag ``--legacy-classes`` will disable this
+new behavior.
+
+Supposing that we have a ``class C`` declaration as in the following:
+
+.. code-block:: chapel
+
+  class C {
+    var x:int;
+  }
+
+Code using ``C`` might refer to the type ``C`` on its own or it might use
+a decorator to specify memory management strategy, as in ``borrowed C``.
+
+The type expression ``C`` was the same as ``borrowed C`` in 1.18 and
+1.19 but now means generic management. For example, in the following code:
+
+.. code-block:: chapel
+
+  var myC:C = new owned C();
+
+``myC`` previously had type ``borrowed C``, and was initialized using
+including an implicit conversion from ``owned C`` to ``borrowed C``. In 1.20,
+``myC`` has type ``owned C``. Since the variable's type expression is
+generic management, it takes its management from the initializing
+expression.
+
+This change combines with the nilability changes described above
+to prevent compilation of existing code like the following:
+
+.. code-block:: chapel
+
+  var x:C;
+
+The 1.20 compiler will find two problems with this statement:
+
+ * ``x`` cannot be default initialized since it cannot store ``nil``
+ * ``x`` cannot be default initialized since it is generic
+
+To update such a variable declaration to 1.20, it is necessary to include
+a memory management decorator. For example:
```

Comment:
"... to include a memory management decorator" and a ``?`` for nilability, right?